### PR TITLE
Catch unwind file indexing iterator

### PR DESCRIPTION
**Pull request by rsdy**
If we allow the file processing to crash, it will gradually eliminate all Rayon worker threads. Log these errors, but don't let Rayon threads die.

### CODE REVIEW

File: server/bleep/src/periodic/remotes.rs [hunk 1]
```diff
@@ -48,11 +48,10 @@ async fn sleep_systime(duration: Duration) {
         return tokio::time::sleep(duration).await;
     }
 
-    let start = SystemTime::now();
-
+    let start = Instant::now();
```
Comment (calyptobai) [inline server/bleep/src/periodic/remotes.rs:51]: As discussed last week, we want to rely on `SystemTime` here as time tracked by `Instant` will not pass on some systems when the OS is suspended.

**Review by calyptobai — COMMENTED** [review]
Was this superseded by #1052?

**Comment by rsdy** [comment]
@calyptobai I didn't notice how dirty this patchset was. In reality, this is supposed to be 1 patch. Updated now.

**Review by calyptobai — APPROVED** [review]
LGTM :+1:

One thing: I think the commit message is slightly misleading here as it says "but don't let Rayon threads die" - this might be outdated text?

**Pull request merged**